addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -316,6 +316,11 @@ export class OdooEditor extends EventTarget {
         // Set contenteditable before clone as FF updates the content at this point.
         this._activateContenteditable();
 
+        this._currentStep = {
+            selection: {},
+            mutations: [],
+        };
+
```

Comment:
I assume the change above is related to the history observation issue but it is not clear to how, and the commit messages provides no explanation. Please update the commit message or add a comment, whichever feels more relevant to you with respect to the actual reason the change was needed.

Reply:
Yeah it's because we call `_setLinkZws` just below (still in the constructor), which triggers the chain `_observerUnactive` (sic) > `observerFlush` > `observerApply` which depends on `_currentStep` having been initialized (`this._currentStep.mutations.push`).

I'll adapt the commit message.
```
Due to these changes, we need to initialize `_currentStep` before
setting the original link zws, so that `observerApply` can access it in
the process of pausing the mutations observer.
```